src/domains/profile/pages/Welcome/Welcome.tsx

```diff
@@ -70,13 +70,16 @@ export const Welcome = () => {
 	const handleRequestedAction = (profile: Contracts.IProfile, action: any, password?: string) => {
 		closeSignInModal();
 
+		if (password) {
+			container.bind("State<Profile>", profile);
```

Comment:
Never set container values directly, tap the profile if you want do something to it before it has been restored.

Reply:
@faustbrian it's the memory password that requires the profile and throws exceptions. In this state, the user's password has been successfully verified, and we need to store it in memory to use it later for restoration and have it available (in profile domains). 

Using sdk's `MemoryPassword` doesn't give this flexibility because it requires the profile to be already set in order to set the password (which is not the case in dw because we want restoration to happen in the next tick, not before). 

`Tap` on the other hand "forgets" the profile and unsets password after callback run, and it seems that we cannot safely rely on `MemoryPassword` and make sure password is not removed from some previously called tap.

So practically we need to set the password and make sure it's there before signing out or calling `env.profiles().forget`. And it needs to happen before restoration.

If `MemoryPassword` is not intended to be used by the consumer, but only internally, we could store active profile's password somewhere else outside sdk, but I wanted to check with you first because it's an important decision.

Reply:
https://github.com/ArkEcosystem/platform-sdk/pull/1395 does this resolve all your issues?

Reply:
Should be ok if it's independent of `State` and can be called prior to profile restoration. Will make a few iterations and send an update. Thanks

Reply:
https://github.com/ArkEcosystem/platform-sdk/releases/tag/6.0.9 should be published in a few